compiler/rustc_typeck/src/check/method/probe.rs

```diff
@@ -1359,6 +1362,24 @@ impl<'a, 'tcx> ProbeContext<'a, 'tcx> {
         })
     }
 
+    fn should_emit_unstable_name_collision_hint(
+        &self,
+        unstable_candidates: &[(Candidate<'tcx>, Symbol)],
+    ) -> bool {
+        !self.is_suggestion.0
+            && !unstable_candidates.is_empty()
+            && unstable_candidates.iter().any(|(candidate, _)| {
+                let stab = self.tcx.lookup_stability(candidate.item.def_id);
+                !matches!(
+                    stab,
+                    Some(Stability {
+                        level: StabilityLevel::Unstable { collision_safe: true, .. },
```

Comment:
It might make sense to make this condition more strict, so we check that `collision_safe` can only prevent the lint from firing *if* the method that has been picked comes from `std`.

Reply:
Fixed this!